src/installer/_core.py

```diff
@@ -69,9 +72,11 @@ def _determine_scheme(path, source, root_scheme):
 
 
 def install(source, destination, additional_metadata):
-    # type: (WheelSource, WheelDestination, Dict[str, Binary]) -> None
+    # type: (WheelSource, WheelDestination, Dict[str, Binary]) -> Mapping[str, Iterable[RecordEntry]]
```

Comment:
I'm a strong -1 on this.

I'd much rather change the interface for the `WheelDestination.finalise_installation` than add a return type to this function. The literal job of `WheelDestination.finalise_installation` is to do things like this.